calicoctl/commands/ipam.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/tigera/libcalico-go/lib/client"
+	caliconet "github.com/tigera/libcalico-go/lib/net"
+
+	docopt "github.com/docopt/docopt-go"
+)
+
+// IPAM show/release is for managing Calico IPAM by showing the
+// current use of an IP and/or releasing the IP address from use
+func IPAM(args []string) error {
+	doc := `Usage: 
+    calicoctl ipam release <IP>
+    calicoctl ipam show <IP>
+
+Description:
+    Manage Calico assigned IP addresses
+
+Warnings:
+  -  Releasing an in-use IP address can result in it being assigned to multiple
+     workloads.
+`
+
+	parsedArgs, err := docopt.Parse(doc, args, true, "calicoctl", false, false)
+	if err != nil {
+		return err
+	}
+
+	// length of parsedArgs is 0 when `-h` `--help` is passed
+	// docopts takes care of printing the help message, but we need to
+	// stop return nil here so command doesn't go any further when `-h` is passed
+	if len(parsedArgs) == 0 {
+		return nil
+	}
+
+	// Parse the input string as an IP address (IPv4 and IPv6)
```

Comment:
There are only two commands at the moment, each of which expects the second arg to be the IP address.  However, this may not always be the case.  I think argument parsing like this should be the responsibility of the subcommand, yeah?